frame/im-online/src/migration.rs

```diff
@@ -0,0 +1,193 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Storage migrations for the im-online pallet.
+
+use super::*;
+use frame_support::{
+	storage_alias,
+	traits::{ConstU32, OnRuntimeUpgrade},
+};
+
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+
+/// The log target.
+const TARGET: &'static str = "runtime::im-online::migration::v1";
+
+/// The original data layout of the im-online pallet (`ReceivedHeartbeats` storage item).
+mod v0 {
+	use super::*;
+	use frame_support::traits::WrapperOpaque;
+
+	/// A type that is the same as `OpaqueNetworkState` but with [`Vec`] replaced with
+	/// [`WeakBoundedVec<Limit>`] where Limit is the respective size limit
+	/// `PeerIdEncodingLimit` represents the size limit of the encoding of `PeerId`
+	/// `MultiAddrEncodingLimit` represents the size limit of the encoding of `MultiAddr`
+	/// `AddressesLimit` represents the size limit of the vector of peers connected
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, MaxEncodedLen, TypeInfo)]
+	pub struct BoundedOpaqueNetworkState<
+		PeerIdEncodingLimit,
+		MultiAddrEncodingLimit,
+		AddressesLimit,
+	> where
+		PeerIdEncodingLimit: Get<u32>,
+		MultiAddrEncodingLimit: Get<u32>,
+		AddressesLimit: Get<u32>,
+	{
+		/// PeerId of the local node in SCALE encoded.
+		pub peer_id: WeakBoundedVec<u8, PeerIdEncodingLimit>,
+		/// List of addresses the node knows it can be reached as.
+		pub external_addresses:
+			WeakBoundedVec<WeakBoundedVec<u8, MultiAddrEncodingLimit>, AddressesLimit>,
```

Comment:
```suggestion
		/// PeerId of the local node in SCALE encoded.
		pub peer_id: Vec<u8>,
		/// List of addresses the node knows it can be reached as.
		pub external_addresses:
			Vec<Vec<u8>>,
```

No need to use `WeakBoundedVec` and makes the code easier.